File: Euripedes Simões de Paula/Java/15 - CRUD + Springbook + Angular/Aprendendo Full Stack/Conectando/src/main/java/com/euripedes/Conectando/model/Diario.java
package com.euripedes.Conectando.model;

import java.time.LocalDate;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class Diario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "credito_id", nullable = false)
    private Conta credito;

    @ManyToOne
    @JoinColumn(name = "debito_id", nullable = false)
    private Conta debito;

    private Double valor;
    private LocalDate data;
    private String historico;

    // Getters e Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Conta getCredito() {
		return credito;
	}

	public void setCredito(Conta credito) {
		this.credito = credito;
	}

	public Conta getDebito() {
		return debito;
	}

	public void setDebito(Conta debito) {
		this.debito = debito;
	}

	public Double getValor() {
        return valor;
    }

    public void setValor(Double valor) {
        this.valor = valor;
    }

    public String getHistorico() {
        return historico;
    }

    public void setHistorico(String historico) {
        this.historico = historico;
    }

    public LocalDate getData() {
        return data;
    }

    public void setData(LocalDate data) {
        this.data = data;
    }
}

File: Euripedes Simões de Paula/Java/16 - Trabalho de Conclusão de Curso/Aprendendo Full Stack Angular/Conectando/src/app/model/Diario.ts
import { Conta } from "./Conta";
import { Historico } from "./Historico";

export class Diario {
    // public id!: number;
    // public data!: Date;
    // public historico!: Historico;
    // // public razoes!: Razao;
    // public valor!: number;
    // public credito!: { id: number } //Conta;
    // public debito!: { id: number } //Conta;

    id!: number;
    credito!: { id: number; };
    debito!: { id: number; };
    data!: string;
    historico!: string;
    valor!: number;
    transacao!: string;
}